src/plugins/intel_cpu/src/nodes/strided_slice.cpp

```diff
@@ -266,11 +317,79 @@ void StridedSlice::initSupportedPrimitiveDescriptors() {
     }
 }
 
+void StridedSlice::resolveInPlaceEdges(Edge::LOOK look) {
+    if (!(look & Edge::LOOK_UP) || !isInPlace()) {
+        Node::resolveInPlaceEdges(look);
+        return;
+    }
+
+    const NodeDesc* selected_pd = getSelectedPrimitiveDescriptor();
+
+    if (!selected_pd)
+        OPENVINO_THROW("Cannot find selected primitive descriptor for node: ", getName());
+
+    auto outputConfig = selected_pd->getConfig().outConfs;
+
+    for (size_t i = 0; i < outputConfig.size(); i++) {
+        auto inplaceInpIndx = outputConfig[i].inPlace();
+
+        if (inplaceInpIndx < 0)
+            continue;
+
+        auto baseMemBlock = getParentEdgeAt(inplaceInpIndx)->getMemory().getMemoryBlock();
+        auto memBlock = std::make_shared<PartitionedMemoryBlock>(baseMemBlock);
+        const auto& childEdges = getChildEdgesAtPort(i);
+
+        auto baseMemorySize = inputShapes.front().getElementsCount();
+
+        // Example:
+        // Input shape: {5, 2, 2, 2}
+        // begin:       {3, 0, 0, 0}
+        // end:         {5, 2, 2, 2}
+        // offset = inputShape[3] * inputShape[2] * inputShape[1] * begin[0]
+        // offset = 2 * 2 * 2 * 3 = 24
+        size_t offset = 1;
+        for (int i = attrs.begin.size() - 1; i >= 0; i--) {
+            if (attrs.begin[i] == 0 && i != 0) {
+                offset *= inputShapes[0].getDims()[i];
+            } else {
+                offset *= attrs.begin[i];
+                break;
+            }
+        }
+
+        size_t end = std::accumulate(attrs.end.begin(), attrs.end.end(), 1, std::multiplies<int>());
+        size_t partitionSize = end - offset;
+
+        for (auto& childEdge : childEdges) {
+            if (childEdge->getStatus() != Edge::Status::NotAllocated) {
+                break;
+            }
+
+            OPENVINO_ASSERT(childEdge->getStatus() == Edge::Status::NotAllocated,
+                            " Unexpected inplace resolve call to an allocated edge: ",
+                            childEdge->name());
+
+            auto memBlock = std::make_shared<PartitionedMemoryBlock>(baseMemBlock, baseMemorySize, offset, partitionSize);
```

Comment:
This memory block can be created only once outside the for loop and just shared across all the peer edges.